new_course/1_programming_fundementals/debugging.md

```diff
@@ -125,9 +125,31 @@ end
 double_words_in_phrase("This is a test")
 ~~~
 
-When your code executes and gets to `binding.pry`, it will open an IRB-like session in your terminal. You can then use that session to check the values of anything within the scope of where you included `binding.pry`.
+When your code executes and gets to `binding.pry`, it will open an IRB-like session in your terminal. You can then use that session to check the values of anything within the scope of where you included `binding.pry`. However, keep in mind that any code written *after* the `binding.pry` statement will not have been evaluated during the Pry session. In other words, when we add a `binding.pry` statement, we are effectively creating a breakpoint or a pause in our code.
 
-Using Pry for debugging achieves the same outcome as `puts` debugging: it allows you to confirm the assumptions you have about particular parts of your code. If your code is complex, Pry will probably allow you to debug quicker thanks to its interactive runtime environment. In such scenarios, Pry will be easier to interact with than having to add `puts` statements everywhere and re-running your code each time.
+To see this point in action, try running the following:
+
+~~~ruby
+# script.rb
+
+require 'pry'
+
+def yell_greeting(string)  
+  name = string
+
+  binding.pry 
+
+  name = name.upcase
+  greeting = "WASSAP, #{name}!"
+  puts greeting
+end
+
+yell_greeting("bob")
+~~~
+
+During the session, if you check for the value of `name`, you will notice that you get back the value `bob` instead of `BOB`. What value do you think `greeting` will return? Yup, it will be `nil`.  This is because `name = name.upcase` and `greeting = "WASSAP, #{name}!"` occurred after the `binding.pry` call and were never evaluated.   
```

Comment:
Looks like there's two spaces between the sentences in this paragraph. Could you please update that to one space to be consistent?